ddl/ddl_api.go

```diff
@@ -5264,6 +5264,9 @@ func (d *ddl) CreateForeignKey(ctx sessionctx.Context, ti ast.Ident, fkName mode
 	if err != nil {
 		return errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(ti.Schema, ti.Name))
 	}
+	if t.Meta().TempTableType != model.TempTableNone {
+		return ErrOptOnTemporaryTable.GenWithStackByArgs("foreign key")
```

Comment:
Please be compatible with MySQL.
`ERROR 1215 (HY000): Cannot add foreign key constraint`